public/app/plugins/panel/heatmap/fields.ts

```diff
@@ -98,15 +98,25 @@ export function prepareHeatmapData({
   });
 
   if (options.calculate) {
-    if (config.featureToggles.transformationsVariableSupport) {
-      const optionsCopy = {
-        ...options,
-        calculation: {
-          xBuckets: { ...options.calculation?.xBuckets } ?? undefined,
-          yBuckets: { ...options.calculation?.yBuckets } ?? undefined,
+    let optionsCopy: Options = {
```

Comment:
we prob just need to do this under the variable support feature flag. the default value of these fields doesn't get applied to the options that are passed in here, look into why that isnt happening